src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -115,9 +124,9 @@ public DistributedApplication Build()
     /// <inheritdoc />
     public IResourceBuilder<T> AddResource<T>(T resource) where T : IResource
     {
-        if (Resources.FirstOrDefault(r => r.Name == resource.Name) is { } existingResource)
+        if (Resources.FirstOrDefault(r => string.Equals(r.Name, resource.Name, StringComparison.OrdinalIgnoreCase)) is { } existingResource)
         {
-            throw new DistributedApplicationException($"Cannot add resource of type '{resource.GetType()}' with name '{resource.Name}' because resource of type '{existingResource.GetType()}' with that name already exists.");
+            throw new DistributedApplicationException($"Cannot add resource of type '{resource.GetType()}' with name '{resource.Name}' because resource of type '{existingResource.GetType()}' with that name already exists. Resource names are case-insensitive.");
```

Comment:
We have a bunch of other code that assumes resource names are case sensitive. Where does the case-insensitivity come from?

Reply:
Oh, I see this in @DamianEdwards's comment on the linked issue. Is that case-insensitivity modelled throughout the whole stack? There's a bunch of code in the dashboard that assumes `StringComparison.Ordinal` for resource names.

Reply:
#1710

Reply:
Resource names are really only case-insensitive when building the model. A case-insensitive check is used to prevent duplicates ("TEST" and "test"). Only one is allowed and will make it into the dashboard.

If there is only "TEST" in the dashboard then it doesn't matter whether case is ignored at that point or not.

Reply:
But a consistent comparer doesn't hurt (well, it's a tiny bit slower comparison when ignoring case, but that will never be impactful)